### ISSUE TITLE
Use alternate to lorem ipsum

### ISSUE THREAD

**Issue by omega-inspiron** [benkeen/generatedata#289]
Would it be possible to have an alternate to lorem ipsum for the random text generator...I'm willing to give a go at creating the plugin if given some guidance that is if one doesn't already exist somewhere.

**Comment by benkeen** [comment]
That's a great idea. Maybe an optional textarea field in the UI for the "Fixed # words", "Random # Words" Data Types to let you can copy + paste in any text you want to use as a source.

The way I wrote it wasn't great, actually. For some "clever" reason I decided to store it in the database, which is really quite unnecessary: it could just be a separate file, or even in the PHP. I think my original reasoning was that multiple Data Types would need it, so I should store it in a single location. 

But anyway, if you fancy tackling this, maybe leave the lipsum text in the DB and just add a new (preferably hidden) field in the UI to let people copy + paste text in. If that's defined, the server-side code would use that instead of the lipsum text. Should be very doable. I can look at it if you'd like, but if you fancy a go - go for it! :)

**Comment by omega-inspiron** [comment]
Haha, I was just looking at the code after I wrote the post to try to figure out if you were pulling the lorem ipsum data set from an api. But alas like you mentioned I found it in the settings table in the database.  I found a few alternates to lorem ipsum here: http://idsgn.dropmark.com/107?page=1.

Presently, maybe the database method is not such a bad idea.  I was thinking along the lines of a hybrid of what you proposed i.e. have a field which allows the user to enter a data set, but then have the ability to store that data set to the db.  The next time its needed, when selecting "Random number of words" etc from the data type show a drop down with a list of data sets that were saved to the db along with a custom data set to continue adding more. (Maybe also set a slot limit so only so many data sets 5 - 10 can be stored at a time)  This is just a thought.  If you have the tiime to look at it that'd be great.

**Comment by omega-inspiron** [comment]
Actually I was further looking at the Utils class and saw this comment above the getLipsum function

	/**
	 * Returns an array of lorem ipsum words. Assumes that a file exists in a misc/ subfolder called
	 * loremipsum.txt, containing lorem ipsum text.
	 *
	 * @return array a large array of words
	 */

Is this something you were thinking about doing but then decided to go the database route?  For now  for what I'm needing I will manually add the data sets to the db manually in a new table, and then write a new function to randomly pull from the data set table.  Till I have the time to modify the front end that is.

**Comment by benkeen** [comment]
Ah - I think that's just an old comment... I guess originally I had it as a separate file. Go figure.

>  I was thinking along the lines of a hybrid of what you proposed i.e. have a field which allows the user to enter a data set, but then have the ability to store that data set to the db.

Nice idea. It's just getting a little fiddly - and very custom: it would mean a core change to satisfy a Data Type, which is kinda weird. To do it well, it would be a fair chunk of work... hmm. 

How about an alternative to your alternative: add a new dropdown in the Random/Fixed Data Type rows in the UI that contains a bunch of preset lipsum alternatives, like those on that site you mentioned. There would also be one final "custom" option which would show a textarea to let you enter your own. You'd get quite a bit of bang for your buck this way. The one limitation is that if you entered manual text into one field, you wouldn't be able to just re-use it in another like with your suggestion. But yeah, I think that would be a fair bit of work.

**Comment by omega-inspiron** [comment]
Yeah your alternate is a suitable compromise.  Thanks.  I'll have a go at this on the weekend.

**Comment by benkeen** [comment]
Nice! Let me know if you get stuck on anything. I'm out of town for a few days on Sat, but should be around internet now and then... :)

**Comment by omega-inspiron** [comment]
OK Will Do.
